Update documentation

naftalimurgor opened the pull request:
Update the documentation as per this issue https://github.com/aromalanil/react-rhino/issues/1
Suggestions are welcome :confetti_ball: 
Thanks

naftalimurgor committed:
Update documentation

update documentation in README.md

naftalimurgor committed:
Merge pull request #1 from naftalimurgor/update-docs

Update documentation as per https://github.com/aromalanil/react-rhino/issues/1

naftalimurgor commented:
Hey @aromalanil 
Let me know your feedback. We can do some changes too.

aromalanil commented on README.md:
````diff
@@ -19,27 +19,17 @@ npm install react-rhino
 yarn add react-rhino
 ```
 
-## Why?
````
I prefer to keep the Why section in the top itself

aromalanil commented on README.md:
```diff
@@ -19,27 +19,17 @@ npm install react-rhino
 
-## Online Playground
-[Here](https://codesandbox.io/s/react-rhino-example-svv5b) is a demo project for you to get familiar with React Rhino.
+1. ### Create global state
```
I see how a step by step instruction would do better, but here the hierarchy of the Heading tags are not correct .

eg: Both "Getting started" & "Create global state" are h3 tags

aromalanil commented on README.md:
```diff
@@ -19,27 +19,17 @@ npm install react-rhino
 
-## Example
-Here is an example of how to use React Rhino
 ### A file to create global states
```
```suggestion
```

I think this line is no more needed

aromalanil commented on README.md:
````diff
@@ -130,11 +131,36 @@ const Toggle = () => {
+}
+```
+
+## Online Playground
````
I think `Online Playground` deserves a better position at the top of the document

aromalanil commented on README.md:
````diff
@@ -150,7 +176,7 @@ const { RhinoProvider, useRhinoState } = createRhinoState({
 ```
 
-> `createRhinoState` will output `RhinoProvider` and a bunch of hooks. Destructure the Provider and the hooks you need.
+> `createRhinoState` will output `RhinoProvider` and `useRhinoState`, `useRhinoValue` and `useSetRhinoState` hooks. Destructure the `createRhinoSate` Provider to get these  hooks as needed(like in the above snippet).
````
```suggestion
> `createRhinoState` will output `RhinoProvider` , `useRhinoState`, `useRhinoValue` and `useSetRhinoState` hooks. 
```

aromalanil commented on README.md:
```diff
@@ -179,21 +205,38 @@ const [darkMode, setDarkMode] = useRhinoState("isDarkMode");
-Having an updater function in the component will not trigger a rerender on the state change 
+> You can use this if your component only needs the updater function and not the state itself.
+Having an updater function in the component will not trigger a rerender on the state change.
+Updator functions perform updates on the state values.
```
```suggestion
```

aromalanil commented:
Hey @naftalimurgor 
Sorry for the delay. My exams are going on 😅

I have added some suggestions, go through each suggestion and change them if we share the same thoughts, if not comment your opinion.

naftalimurgor commented:
The suggestions are perfectly great. Will work on them. Btw, I haven't used markdown that much. I wish you success in your exams ! 😃

naftalimurgor committed:
Correct `h3` to `h4`   tags for step-by-step guide

change h3 tags to h4 for the steps to have a correct hierarchy in the step by step guide

naftalimurgor committed:
Place "why choose Rhino" section at the top

naftalimurgor committed:
Update documentation sections

Make overall changes to documentation sections.

naftalimurgor committed:
Add License type to docs

naftalimurgor committed:
Merge pull request #2 from naftalimurgor/updated-docs

Updated docs

naftalimurgor commented:
Hey @aromalanil 
I made the changes per the suggestions we had. I also added License type at the bottom :smile: Let me know if there are other enhancements we can add.
Thanks!

aromalanil commented:
@naftalimurgor Thanks for the changes. Just a few more small corrections and this will be merged.

1. Installation should come just before why section.
2. There should only be 3 steps in the getting started section. Others can be added as separate heading under usage section with same hierarchy as Getting started

naftalimurgor commented:
@aromalanil 
Great! I'll make the changes to the mentioned sections.

naftalimurgor committed:
Update documentation

update usage section and change steps to 3 steps

naftalimurgor committed:
Merge pull request #3 from naftalimurgor/updated-docs

Update documentation

aromalanil commented:
LGTM 🌷

Congrats @naftalimurgor on your first merge to React Rhino
![Celebrate](https://giphy.com/gifs/celebration-excited-loki-vmon3eAOp1WfK)

aromalanil commented:
LGTM 🌷

Congrats @naftalimurgor on your first merge to React Rhino
![Celebrate](https://user-images.githubusercontent.com/49222186/125281909-6b201000-e334-11eb-8a1d-ffdc5118610d.gif)
